code/modules/minimap/minimap.dm

```diff
@@ -141,34 +154,98 @@
 			max_y = max(max_y, T.y)
 			min_y = min(min_y, T.y)
 
+		var/focus_x = ((max_x + min_x) - 1) / 2
+		var/focus_y = ((max_y + min_y) - 1) / 2
+
 		var/scale_x = world.maxx / ((max_x - min_x) + border_width)
 		var/scale_y = world.maxy / ((max_y - min_y) + border_width)
-		src.zoom_coefficient = min(scale_x, scale_y)
 
-		src.focus_x = ((max_x + min_x) - 1) / 2
-		src.focus_y = ((max_y + min_y) - 1) / 2
+		src.centre_on_point(min(scale_x, scale_y), focus_x, focus_y)
+
+	///Zooms the minimap by the zoom coefficient while moving the minimap so that the specified point lies at the same position on the displayed minimap as it did prior to the zoom. The alpha mask takes care of any map area scaled outside of the map boundaries.
+	proc/zoom_on_point(var/zoom, var/map_x, var/map_y)
+		if (!zoom || zoom < 1 || zoom > 20 || !map_x || !map_y)
```

Comment:
Would that not result in redundant zoom calculations being performed?

Reply:
it would, it's fine to leave if you think it matters, but it might be nice to turn that 20 into a var/define at least (since it's used in the other proc too)